.github/workflows/pr_and_main.yml

```diff
@@ -40,7 +40,10 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        variant:
+        - {os: ubuntu-20.04, cmake-preset: ci-linux}
+        - {os: macos-10.15, cmake-preset: ci-macos}
+        - {os: windows-2019, cmake-preset: ci-windows}
```

Comment:
Is this the latest?

Reply:
No.

Reply:
Why did you use it?

Reply:
Agreed, should be `windows-2022`? Moving back to the older version could potentially break some things like `ccache` if it was relying on some recent windows features?

Reply:
Please see the PR description. `windows-2022` is the one that broke our builds.

Reply:
But looking at the table here: https://github.com/actions/virtual-environments It seems like `ubuntu-latest` is also just an alias to `ubuntu-20.04`. I don't see the difference.
Also, it does not look like they keep all versions around, e.e.g `ubuntu-20.03`. So what happens when they update to `20.05`? Would this break all our builds?

Reply:
Yes I also noticed the 20.04/latest alias. But:
- Ubuntu-20.04 is a distribution and should not see updates apart from security.
- Since we use gcc from conda even hypothetical breaking compiler updates should not affect us.

Regarding your concerns without image updates: They announce this long in advance, so we will have the chance to update.

Reply:
> Also, it does not look like they keep all versions around, e.e.g ubuntu-20.03. So what happens when they update to 20.05? Would this break all our builds?

I don't think `20.03` exists. The numbers code for year and month. `20.04` means it was released in April of 2020. It then goes `20.10` (october), `21.04`, `21.10`, 22.04` etc..

Reply:
> windows-2022 is the one that broke our builds.
Sure, but it is not clear whether they do see updates or not. The PR message says "hope"

Reply:
By "they", what do you refer to?

Reply:
the old images

Reply:
Well, we have to use *some* image. We *know* that the new ones are updated, so that is not a good option. Do you suggest providing our own image?

Reply:
> Do you suggest providing our own image?

No, I just wanted to make sure we thought about the different implications of changing back to an old image. If there were e.g. some C++ features that we use that were not available (obviously this would fail at compilation time), or the ccache (as mentioned above). I don't see any specification on the ccache docs about which versions of MSVC compilers are supported, so it's probably fine with 2019.

I'm not bothered what the actual version number is, as long as everything works ;-)

Reply:
That is why I made a PR, so we can see if the builds pass.